10.12.0 rn

MariaShaposhnikova opened the pull request.

MariaShaposhnikova committed:
Add an empty draft

MariaShaposhnikova committed:
Add 10.12.0 release notes + review

MariaShaposhnikova committed:
Add TW notes

quinntracy committed:
Revision

sbont commented:
We'd like to promote _features_ on lines 41 & 42 to it's own section, as follows:

```markdown
#### Using Associations with External Actions
**Published OData services** now support including associated objects when calling a microflow that is published as an OData action. When consuming these actions in a **Call External Action** activity, Studio Pro lets you select the associations that should be included in the action call.  These associated objects are then included in the request to the **Consumed OData service**. If the service returns an object or a list of objects, you can also specify which associations you want to get back from the service. These associations can then be traversed using the __Retrieve__ activity. 
If you call the published OData service using a REST client, you can nest the associated objects in the request body, and you can use the `$expand` query parameter to control which associated objects should be returned in the response.
```

MarkvanMents commented on content/en/docs/releasenotes/studio-pro/10/10.12.md:
```diff
@@ -0,0 +1,146 @@
+- We added an **Managed Dependency Exclusions** section in the **Dependencies** tab of the **App settings**. In this section you can exclude managed dependencies from the app deployment. Use this if there are conflicts between dependencies in your application.
+- You can now configure exclusions for managed dependencies in the **Module Settings**. Using this feature you can exclude certain transitive dependencies from a managed dependency to fix conflicts and ensure these transitive dependencies will not be included in your project.
+### Content-Security-Policy with nonces
+Previously, you have been able to define the Content-Security-Policy (CSP) for your apps in the Cloud Portal. However, this would only allow for basic policies. With this release, we have added the ability to directly define the CSP header as a custom runtime configuration. {TW}
```
Cloud Portal?

ConnorLand replied:
I discussed this docs addition in Slack under Maria's RN post for it. Basically, can you look into if [this doc](https://docs.mendix.com/howto/security/csp/) or Cloud docs need an update as to this new custom runtime configuration?

Robinfr replied:
I just created a new PR to address the documentation around the new Content Security Policy changes.

Robinfr replied:
Ah I think the confusion here is about "Cloud portal" tho. Apparently, it's called "Developer portal" nowadays..

Robinfr replied:
After internal discussion, we unfortunately have to remove this feature from the release note. @ConnorLand @MariaShaposhnikova can you please remove it for now?

MariaShaposhnikova replied:
@Robinfr , removed! Thanks for the update!

MariaShaposhnikova committed:
High-level review of fixes

MariaShaposhnikova committed:
Merge branch '10.12.0-rn' of https://github.com/mendix/docs into 10.12.0-rn

MariaShaposhnikova committed:
Implement a suggestion in comment

blis-mendix commented on content/en/docs/releasenotes/studio-pro/10/10.12.md:
```diff
@@ -0,0 +1,148 @@
+
+For more information on how to set up CSP, see [Content Security Policy](/howto/security/csp/).
+
+### Improvements
```
Hey @MariaShaposhnikova sorry for the trouble but we (MCR) are missing one of the improvements we did, partial cloning could you add this to the release notes: 

We added a support for using partial application clones with Git, instead of full clones. Partial clones download a minimum required set of data, due to which it is significantly faster when working with large repositories. Through the app launcher settings or version control preferences you can choose which cloning strategy to use for future cloning operations, such as downloading a new app or branch.

tknippenberg replied:
Suggest to mention that "we added a public beta for ..." so it's clear it's beta.

ConnorLand replied:
Added your note, made it clear that it's in public beta. Nice feature! Thanks for the note.

tknippenberg replied:
Can you add this link: https://docs.mendix.com/refguide/clone-type/ to the word "partial application clones" (from this [PR](https://github.com/mendix/docs/pull/7853))?

I think it could also make sense to add the link https://docs.mendix.com/refguide/troubleshoot-repository-size/ on "large repositories".

ConnorLand replied:
Done.

MariaShaposhnikova replied:
@tknippenberg , we also need a note for the correspoding document for Partial Cloning. I will add it, but just FYI.

ConnorLand committed:
Add Partial Clone improvement

Via docs slack suggestion

ConnorLand committed:
Add partial clone links

ConnorLand committed:
re-order fixes by ticket number

ConnorLand committed:
Align Fix text with previous RNs

ConnorLand committed:
Fix for multi-user workflow KI

ConnorLand committed:
Change KI fix anchor link

Also fixed a typo in the deprecations header

MariaShaposhnikova committed:
Review until fixes

MariaShaposhnikova commented on content/en/docs/releasenotes/studio-pro/10/10.12.md:
```diff
@@ -0,0 +1,149 @@
+
+- We moved the XPath Visual Builder is now GA (it was in public beta).
+
+- Selection can now be modified by the CTRL/CMD key, either by a single click or by dragging a selection rectangle, allowing for more precise control in selecting elements in logic flow editors.
```
This needs a button formatting.

ConnorLand replied:
Done.

MariaShaposhnikova commented on content/en/docs/releasenotes/studio-pro/10/10.12.md:
```diff
@@ -0,0 +1,149 @@
+- In Java actions, optional parameters without mappings will no longer trigger a consistency error (CE0115).
+- We added an improvement (public beta) which allows you to use [partial application clones](/refguide/clone-type/) with Git instead of full clones. Partial clones download a minimum required set of data, making this method significantly faster than full clones when working with [large repositories](/refguide/troubleshoot-repository-size/). Through **Open app** settings or version control prefere ...
+- We added an improvement where if you use parameters in the body of a Consumed REST Request (beta), the parameters that are not defined yet get added automatically to the parameter tab with empty test values.
+- We optimized data retrieval in [Optimize for Network Load mode](/refguide/datasource-runtime/#32-optimize-for-network-load) even further. Entity paths in expressions now only retrieve the ID attributes of associated objects.
```
This is not a correct link, probably needs an anchor first in the doc.

ConnorLand replied:
Fixed.

quinntracy committed:
Revise OData notes

ConnorLand committed:
Remove KI from metadata of previous versions

ConnorLand committed:
redo KI link titles

ConnorLand committed:
Fix anchor and formatting

ConnorLand commented on content/en/docs/releasenotes/studio-pro/10/10.12.md:
```diff
@@ -0,0 +1,151 @@
+
+The advantage is that microflows and nanoflows can now more easily be (re)used when they have primitive parameters, as they can be called directly from a page. Previously, multiple microflows or nanoflows had to be used, or a non-persistent entity had to be introduced in both the logic and UI. Not anymore!
+
+For example, there are multiple buttons to change the status of an attribute. Before this feature, this would require one microflow per button. Now, just one microflow can be used that has the status as primitive parameter and pass that from the button. Plain and simple. 
```
I think this sentence needs to be clarified with the dev. Its tense changes and prepositions make its meaning hard to parse. 

"Now, just one microflow can be used that has the status as primitive parameter and pass that from the button."

ancajuravle replied:
I'll discuss it with the team to see if we can make it clearer.

weirdwater replied:
How about making the example concrete, like this:

> For example, picture a page allowing the user to approve or deny a request using two separate buttons. Before this feature, each button would require their own microflow. Now both buttons can call the same microflow passing the status as a primitive argument. Plain and simple.

weirdwater replied:
This example is based on this request from the community: https://community.mendix.com/link/space/studio-pro/ideas/1170

ConnorLand replied:
Perfect example and nicely written. Thanks @weirdwater !

ConnorLand replied:
I gave credit to the user @weirdwater indicated in his link here: https://github.com/mendix/docs/pull/7898/commits/9a8bd5faaf0b17df527640eb32d36d76570670ff

ConnorLand committed:
Add changes from PR #7903

I also changed some dashes to asterixes.

ConnorLand committed:
Edit language

If I changed any ticketed fixes, I made sure to check for other SP versions with those same fixes.

MariaShaposhnikova committed:
Add MTS note and label

MariaShaposhnikova committed:
Review

MariaShaposhnikova committed:
Review via Slack

MarkvanMents committed:
Document External Database Connector incompatibility

MarkvanMents commented on content/en/docs/appstore/modules/database-connector-mx10.md:
```diff
@@ -13,6 +13,8 @@ Connect to Microsoft SQL, MySQL, PostgreSQL, Oracle, and Snowflake (Beta support
 
 {{% alert color="info" %}}
 The External Database Connector is supported for [Studio Pro 10.6](/releasenotes/studio-pro/10.6/) and above.
+
+Versions 2.0.1 and below are not compatible with Studio Pro version 10.12 and above. You will need to upgrade to version 3.0.0 or above.
```
Actually, this is covered in #7917 so it can be removed from here.
```suggestion
```

MarkvanMents committed:
Duplicate change from #7917

MariaShaposhnikova committed:
Review via Slack

MariaShaposhnikova committed:
Update link

MariaShaposhnikova committed:
Update latest releases

MariaShaposhnikova committed:
Fix links

MariaShaposhnikova committed:
Remove comments

ConnorLand committed:
Rewrite example

ConnorLand committed:
Added credit for Tim from the Forum

ConnorLand committed:
Give Strict Mode its own section

ConnorLand committed:
Add APPEXT-796 KI

Yiyun333 committed:
Add LFX-712 KI to 10.6-10.12 RNs

Yiyun333 commented on content/en/docs/releasenotes/studio-pro/10/10.12.md:
```diff
@@ -0,0 +1,158 @@
+* Line endings in CSS files are not being handled properly, so when using [Revert All Changes](/refguide/using-version-control-in-studio-pro/) or performing other version-control operations, CSS files appear in the [Changes on Disk](/refguide/version-control-menu/#show-changes) dialog box.
+    * Workaround: For details, see [this section](/refguide/troubleshoot-version-control-issues/#css-error) in *Troubleshooting Version Control*.
+* A finished parallel split path that is removed from a running workflow instance wrongly leads to a versioning conflict. Please note that "path" was called "branch" in earlier versions.
+* When using a microflow decision method for a multi-user task in workflows, the `System.WorkflowUserTask` is always `empty`.
```
@ConnorLand This KI should be deleted. Is this saved as a reminder? I don't think 10.6.11 will be released tomorrow morning before 10.12.0 :)

ConnorLand replied:
This was not left in as a reminder, it it simply wrong. I will delete it, as indeed the fix is noted above. Indeed, as you say, because 10.6.11 will not come out before 10.12, we do not need to worry about altering the link text.

Yiyun333 committed:
Remove empty space and fixed KI

ConnorLand committed:
Merge pull request #7934 from mendix/yl-add-logic-editor-ki-LFX712

Add LFX-712 KI to 10.6-10.12 RNs

Yiyun333 committed:
Add Maia RNs

ConnorLand committed:
Add date to RNs
